infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -321,6 +324,15 @@ private void multiRead(CorfuPayloadMsg<MultipleReadRequest> msg, ChannelHandlerC
         }
     }
 
+    private ILogData readSingle(long address, boolean cacheable) {
+        if (!cacheable) {
+            ILogData ld = dataCache.getIfPresent(address);
+            return ld != null ? ld : handleRetrieval(address);
```

Comment:
Could we add a test to validate, that reads are not cached and read again?

Reply:
Actually I think it would not  be easy to test this part. This needs to separate the cache, and unit test on it.

Reply:
I had the same thought as @annym . 

I think we should wrap the Caffeine cache and only access the LogUnitCache. Also, you can have a method with like `cache.read(key, CacheOptions)`

Reply:
@annym @Maithem Done. Separated data cache and added a test.